### ISSUE TITLE
Division of labor between game and web servers

### ISSUE THREAD

**Issue by hintron** [Jldevictoria/Oldentide#63]
I think we need to think deeply about what we want the web server to do, and what we want the game server to do. I think that **anything that doesn't require real-time response times should be offloaded to the web server**.

The benefits to relying on the web server is that there is no need to check for lost packets, like UDP. There is no need to worry about timing or a protocol. Instead, we can just use it as an HTTP REST API. There is also no worry that a non-essential non-real-time function like loading a player list will degrade critical multiplayer game performance.

For example, we have a 'get player list' UDP packet. However, this action isn't needed in real-time during gameplay. You'll only need to load the character list before gameplay starts. Se we don't need UDP packets like REQCLIST, SENDCLIST, or CREATEPLAYER. There may also be a case for making all communication route through the web server as well, so there are no dropped messages and so messages don't affect game performance.

**Comment by Jldevictoria** [comment]
I agree some things can be moved to the http server, but I am hesitant to agree than anything not time sensitive can be moved.  I would really like to handle as much game traffic as possible in the game server.  

I don't mind handling some of the login procedure through the web, but if we take it one step further, we've essentially created a web game.

**Comment by ajgoda90** [comment]
Having real-time over UDP, and non-realtime over TCP is probably a good choice, but keeping those on "one" server would be good too, especially if there's a desire to help people self-host their own worlds. So while on the inside the UDP and TCP traffic might be handled by distinct processes or threads, in the outside they are run together on the same machine with one command and configuration. Things like authentication and sending stats to the web can be made pluggable, so it's easier to swap out.

**Comment by hintron** [comment]
@ajgoda90 Currently, the web server is implemented as a spawned off thread of the game server, which is very convenient. However, this means that if one of the servers crashes, you'd need to restart both of them. Having to stop and restart the game server to restart the non-real time web server isn't ideal. We shouldn't need to restart a critical service to reboot a non-critical service ("critical" here meaning critical to continuing gameplay for people already logged in and playing in the world). So maybe separate binaries could address this issue better.
